lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunication.cpp

```diff
@@ -1316,6 +1316,7 @@ std::string GDBRemoteCommunication::ExpandRLE(std::string packet) {
     } else {
```

Comment:
I can't comment on it for some reason, but above this else is an `else if (*c == 0x7d)`. That block also does `*++c`, which will also go out of bounds if `std::next(c) == packet.end()`. You need to add a check there too, don't you?